receiver/sqlserverreceiver/scraper.go

```diff
@@ -300,3 +329,231 @@ func (s *sqlServerScraperHelper) recordDatabaseStatusMetrics(ctx context.Context
+			s.logger.Warn("problems encountered getting metric rows", zap.Error(err))
+		} else {
+			return plog.Logs{}, fmt.Errorf("sqlServerScraperHelper failed getting rows: %w", err)
+		}
+	}
+	var errs []error
+
+	totalElapsedTimeDiffs := make([]int64, len(rows))
+
+	for i, row := range rows {
+		queryHashVal := hex.EncodeToString([]byte(row[queryHash]))
+		queryPlanHashVal := hex.EncodeToString([]byte(row[queryPlanHash]))
+
+		elapsedTime, err := strconv.ParseInt(row[totalElapsedTime], 10, 64)
+		if err != nil {
+			s.logger.Info(fmt.Sprintf("sqlServerScraperHelper failed getting rows: %s", err))
+			errs = append(errs, err)
+		} else {
+			// we're trying to get the queries that used the most time.
+			// caching the total elapsed time (in millisecond) and compare in the next scrape.
+			if cached, diff := s.cacheAndDiff(queryHashVal, queryPlanHashVal, totalElapsedTime, elapsedTime/1000); cached && diff > 0 {
+				totalElapsedTimeDiffs[i] = diff
+			}
+		}
+	}
+
+	// sort the rows based on the totalElapsedTimeDiffs in descending order
+	rows = sortRows(rows, totalElapsedTimeDiffs)
+
+	// sort the totalElapsedTimeDiffs in descending order as well
+	sort.Slice(totalElapsedTimeDiffs, func(i, j int) bool { return totalElapsedTimeDiffs[i] > totalElapsedTimeDiffs[j] })
+
+	logs := plog.NewLogs()
+	resourceLog := logs.ResourceLogs().AppendEmpty()
+	resourceLog.Resource().Attributes().PutStr("db.system.type", "microsoft.sql_server")
+
+	scopedLog := resourceLog.ScopeLogs().AppendEmpty()
+	scopedLog.Scope().SetName("github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sqlserverreceiver")
+	scopedLog.Scope().SetVersion("development")
+
+	timestamp := pcommon.NewTimestampFromTime(time.Now())
+	for i, row := range rows {
+		// we skip the query if we already export enough queries in this run.
+		if i > int(topQueryCount) {
+			continue
+		}
+
+		// reporting human-readable query hash and query hash plan
+		queryHashVal := hex.EncodeToString([]byte(row[queryHash]))
+		queryPlanHashVal := hex.EncodeToString([]byte(row[queryPlanHash]))
+
+		record := scopedLog.LogRecords().AppendEmpty()
+		record.SetTimestamp(timestamp)
+
+		record.Attributes().PutStr(computerNameKey, row[computerNameKey])
+		record.Attributes().PutStr(instanceNameKey, row[instanceNameKey])
+
+		record.Attributes().PutStr(DBPrefix+queryHash, queryHashVal)
+		record.Attributes().PutStr(DBPrefix+queryPlanHash, queryPlanHashVal)
+
+		s.logger.Debug(fmt.Sprintf("QueryHash: %v, PlanHash: %v, DataRow: %v", queryHashVal, queryPlanHashVal, row))
+
+		record.Attributes().PutInt(DBPrefix+totalElapsedTime, totalElapsedTimeDiffs[i])
+
+		// handling `total_rows`
+		rowsReturnVal, err := strconv.ParseInt(row[rowsReturned], 10, 64)
+		if err != nil {
+			err = fmt.Errorf("row %d: %w", i, err)
+			errs = append(errs, err)
+		}
+		if cached, diff := s.cacheAndDiff(queryHashVal, queryPlanHashVal, rowsReturned, rowsReturnVal); cached {
+			record.Attributes().PutInt(DBPrefix+rowsReturned, diff)
```

Comment:
why are you trying to send as an attribute a delta of the value? Why not send this value as is and let the backend perform a query to compare?